### DISCUSSION TITLE
Codifica circoscrizioni

### DISCUSSION THREAD

**Discussion by umbaro** [ondata/elezioni-politiche-2022#23] [Domande e risposte]
Posso chiedervi come avete codificato le diverse circoscrizioni? Faccio difficoltà a capire a cosa corrisponde cosa 🤷🏻‍♂️

**Comment by aborruso** [comment]
Ciao @umbaro e benvenuto. Il Ministero degli interni usa una sua codifica e purtroppo non pubblica in modo più evidente una stele di rosetta.

I dati comunali su Eligendo sono dei JSON con questa struttura

```json
{
    "st": "ESERCIZIO",
    "t_ele": "Camera Italia",
    "f_elet": "SCRUTINI",
    "dt_ele": 20220925000000,
    "l_terr": "COMUNE",
    "area": "I",
    "cod_com": "2620",
    "desc_com": "TORINO",
    "cod_prov": "81",
    "desc_prov": "TORINO",
    "cod_cl_uni": "111",
    "desc_cl_uni": "PIEMONTE 1 - U01 (TORINO: CIRCOSCRIZIONE 2 - SANTA RITA - MIRAFIORI NORD - MIRAFIORI SUD)",
    "cod_plu": "11",
    "desc_cl_plu": "PIEMONTE 1 - P01",
    "cod_circ": "1",
    "desc_circ": "PIEMONTE 1",
    "ele_m": 148679,
    "ele_f": 168856,
    "ele_t": 317535,
    "vot_m": 100864,
    "vot_f": 111411,
    "vot_t": 212275,
    "perc_vot": "66,85",
    "sz_perv": 468,
    "sz_tot": 468,
    "tot_vot_uni": 206086,
    "tot_vot_prop": 198065,
    "fine_rip": "N",
    "sk_bianche": 1461,
    "sk_nulle": 4659,
    "sk_contestate": 69,
    "dt_agg": 20220926073040,
    "data_prec_elez": null,
    "circ_sto": null,
    "cod_plu_sto": null,
    "cod_cl_uni_sto": null,
    "cod_prov_sto": null,
    "cod_com_sto": null
  }
```

Trovi nota di un codice comunale, di un codice provinciale, ecc..

Per rimettere tutto insieme, devi studiarti un po' di fonti. La principale è questa, dove hai un file sui comuni https://dait.interno.gov.it/territorio-e-autonomie-locali/open-data/dati-sistema-unico-territoriale

Altre due fonti sono queste:

- la suddivisione generale territoriale di Eligendo https://github.com/ondata/elezioni-politiche-2022/blob/main/affluenza-risultati/risorse/territorio.csv
- la suddivisione territoriale per la Camera (sempre da Eligendo) https://github.com/ondata/elezioni-politiche-2022/blob/main/affluenza-risultati/risorse/camera_geopolitico_italia.csv

Studiandosi il tutto, riesci a capire le correlazioni. Ma quello che chiedi è utile per tutti, allora abbiamo aggiunto il codice ISTAT comunale nel file di anagrafica dei risultati elettorali della Camera:
<https://github.com/ondata/elezioni-politiche-2022/blob/main/affluenza-risultati/dati/risultati/camera-italia-comune_anagrafica.csv>

Così potrai associargli tutti gli altri codici geografici del caso.

Fammi sapere

**Reply by aborruso** [reply]
@umbaro hai risolto?

Sono iniziate a uscire le prime mappe, basate su quei dati:

- https://gjrichter.github.io/pages/Elezioni_Politiche_2022_scrutini/index_comuni_extended.html
- https://public.flourish.studio/visualisation/11321338/
